src/board/queen.rs

```diff
@@ -35,39 +35,39 @@ mod test {
         let moved_boards = board.generate_moves(true).unwrap();
 
         let expected_moves = vec![
-            Square { rank: 2, file: 3 },
-            Square { rank: 2, file: 2 },
-            Square { rank: 2, file: 1 },
-            Square { rank: 2, file: 0 },
-            Square { rank: 1, file: 4 },
-            Square { rank: 0, file: 4 },
-            Square { rank: 3, file: 4 },
-            Square { rank: 4, file: 4 },
-            Square { rank: 5, file: 4 },
-            Square { rank: 6, file: 4 },
-            Square { rank: 4, file: 1 },
-            Square { rank: 4, file: 0 },
-            Square { rank: 4, file: 3 },
-            Square { rank: 4, file: 4 },
-            Square { rank: 4, file: 5 },
-            Square { rank: 4, file: 6 },
-            Square { rank: 4, file: 7 },
-            Square { rank: 3, file: 2 },
-            Square { rank: 2, file: 2 },
-            Square { rank: 1, file: 2 },
-            Square { rank: 5, file: 2 },
-            Square { rank: 6, file: 2 },
-            Square { rank: 1, file: 3 },
-            Square { rank: 0, file: 2 },
-            Square { rank: 1, file: 5 },
-            Square { rank: 0, file: 6 },
-            Square { rank: 3, file: 3 },
-            Square { rank: 3, file: 5 },
-            Square { rank: 4, file: 6 },
-            Square { rank: 5, file: 3 },
-            Square { rank: 6, file: 4 },
-            Square { rank: 5, file: 1 },
-            Square { rank: 6, file: 0 },
+            UncheckedSquare { rank: 2, file: 3 }.validate(&board).unwrap(),
```

Comment:
Why are we taking squares that _we know are valid_ and are already `Square`s, and then turning them into `UncheckedSquare`s that we then `validate` and `unwrap`... anyway? I'm just very confused here.

Reply:
I did those so that the Square method could have its properties protected from access. I could fix it by creating a function that's only around in a #[cfg(test)] that could create Squares that we know are valid.